tests/tests_user_views.py

```diff
@@ -0,0 +1,124 @@
+        'account_type': Account_type.TEACHER,
+        'phone_number': '+1234567890',
+        'city': 'Test City',
+        'meeting_method': Meeting_method.LIVE
+    }
+
+
+@pytest.mark.django_db
+class TestProfileViews:
+    def test_users_detail_view_loaded_authorized(self, authorized_client, persist_user_profile):
+        users_detail_response = authorized_client.get(f'/users/{persist_user_profile.user.username}/', follow=True)
+        assert users_detail_response.status_code == 200
+        assert 'profile_details.html' in users_detail_response.templates[0].name
+
+    def test_users_detail_view_loaded_unauthorized(self, client, persist_user_profile):
+        users_detail_response = client.get(f'/users/{persist_user_profile.user.username}/', follow=True)
+        assert users_detail_response.status_code == 200
+        assert 'main/login.html' in users_detail_response.templates[0].name
+
+    def test_edit_profile_view_unauthenticated(self, client, new_profile):
+        url = reverse('users-edit-profile', kwargs={'slug': new_profile.user.username})
+        response = client.get(url, follow=True)
+        assert response.status_code == 200
+        assert 'main/login.html' in response.templates[0].name
+
+    def test_edit_profile_view_authenticated(self, client, new_profile):
+        client.login(username='testuser', password='testpassword')
+        url = reverse('users-edit-profile', kwargs={'slug': new_profile.user.username})
+        response = client.get(url, follow=True)
+        assert response.status_code == 200
+        assert 'main/login.html' in response.templates[0].name
+
+    def test_users_update_view_loaded_authorized(self, authorized_client, persist_user_profile):
+        user_detail_response = authorized_client.get(f"/users/{persist_user_profile.pk}/edit-profile/")
+        assert user_detail_response.status_code == 200
+        assert 'edit-profile.html' in user_detail_response.templates[0].name
+
+    @pytest.mark.parametrize("email, city", [(EMAIL, "city_test")])
+    def test_update_user(self, persist_user_profile, authorized_client, email, city):
+
+        profile = Profile.objects.filter(user=persist_user_profile.user).first()
+        assert profile.city != city
+        assert profile.user.email != email
+
+        new_profile = {"username": persist_user_profile.user.username,
+                       "email": email,
+                       "first_name": "",
+                       "last_name": "",
+                       "bio": "",
+                       "profession": "",
+                       "city": city,
+                       "phone_number": '0500000000',
+                       "account_type": 'B',
+                       "meeting_method": "B"}
+
+        response = authorized_client.post(f"/users/{persist_user_profile.user.username}/edit-profile/",
+                                          new_profile, follow=True)
+        profile = Profile.objects.filter(user=persist_user_profile.user).first()
```

Comment:
I don't think this is needed, you already test this part in the model

Reply:
but the test is failed without this line

Reply:
Well, you can't delete just this line... see my next comment

Reply:
The persistent user is not updated automatically, so to check for an update I have to actually pull it out from the database again, that's why this line is necessary